datastore/dbstore/member.go

```diff
@@ -203,3 +203,32 @@ VALUES `
 
 	return err
 }
+
+// ProcessMember - add them member if they don't already exist.  Otherwise, make sure we have their name
+func (db *DatabaseStore) ProcessMember(newMember models.Member) error {
+	member, err := db.GetMemberByEmail(newMember.Email)
+	if err != nil {
+		log.Errorf("error looking up member: %s", err)
+		return err
+	}
+
+	// if email is blank, we didn't retrieve a member from the DB
+	// let's add the member to the db
+	if member.Email == "" {
+		db.AddMembers([]models.Member{newMember})
+		return nil
+	}
+
+	// if we already have a their name, exit
```

Comment:
Wouldn't it better to do this?

` if member.Email == "" && member.Name == "" `

Then you will really know the member doesn't exist and plus I think it's weird you're checking after creating the member.

Reply:
I'll try to rewrite this so that it's more clear.

If the member was added manually, they won't have a name stored in the DB, but they will have an email address.
So, if they don't have an email address, they aren't in the DB and we need to add them.
if they do have an email, but they don't have a name, we should update their name in the db.